module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/MakeTagTest.php

```diff
@@ -0,0 +1,245 @@
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+namespace VuFindTest\View\Helper\Root;
+
+use VuFind\View\Helper\Root\MakeTag;
+
+/**
+ * makeTag view helper Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class MakeTagTest extends \VuFindTest\Unit\AbstractMakeTagTest
+{
+    /**
+     * Get makeTag helper with mock view
+     *
+     * return \Laminas\View\Helper\EscapeHtml
+     */
+    protected function getHelper()
+    {
+        $helper = new MakeTag();
+        $helper->setView($this->getViewWithHelpers());
+        return $helper;
+    }
+
+    /**
+     * Test that responds to common inputs
+     */
+    public function htmlAttributesTests()
+    {
+        return [
+            // Basic
+            [
+                '<button class="btn" id="login">text</button>',
+                ['button', 'text', ['class' => 'btn', 'id' => 'login']]
+            ],
+
+            // String
+            [
+                '<i class="btn">text</i>',
+                ['i', 'text', 'btn']
+            ],
+
+            // Empty text
+            [
+                '<i class="fa&#x20;fa-awesome"></i>',
+                ['i', '', 'fa fa-awesome']
+            ],
+
+            // Truthy attribute
+            [
+                '<a href="&#x2F;login" data-lightbox="1">Login</a>',
+                ['a', 'Login', ['href' => '/login', 'data-lightbox' => true]]
+            ],
+        ];
+    }
+
+    /*
+     * Void elements for test below
+     */
+    public function helperOptionTests()
+    {
+        return [
+            // escapes innerHTML
+            [
+                '<button>This link is &lt;strong&gt;important&lt;/strong&gt;</button>',
+                [
+                    'button',
+                    'This link is <strong>important</strong>',
+                ]
+            ],
+
+            // does not escape innerHTML with option
+            [
+                '<button>This link is <strong>important</strong></button>',
+                [
+                    'button',
+                    'This link is <strong>important</strong>',
+                    [],
+                    ['escapeContent' => false]
+                ]
+            ],
+
+            // escape innerHTML with option
+            [
+                '<button>This link is &lt;strong&gt;important&lt;/strong&gt;</button>',
+                [
+                    'button',
+                    'This link is <strong>important</strong>',
+                    [],
+                    ['escapeContent' => true]
+                ]
+            ],
+        ];
+    }
+
+    /*
+     * Void elements for test below
+     */
+    public function voidTags()
+    {
+        return [
+            // self closing tag
```

Comment:
If a data provider returns an associative array, the keys are the names of the test cases. Rather than putting comments here, it's probably better to use keys -- e.g. `'self closing tag' => [ ... ]`. This way, when a test fails, PHPUnit will tell you exactly which case caused the failure.